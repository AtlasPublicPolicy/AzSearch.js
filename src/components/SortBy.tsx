import * as React from "react";
import { PropsType } from "../containers/SortByContainer";
import * as objAssign from "object-assign";
import { Store } from "azsearchstore";
import { defaultCss } from "../utils/css";

export type State = {};

class SortBy extends React.Component<PropsType, State> {
  render() {
    return (
      <div>Sort!</div>
    );
  }
}

export default SortBy;