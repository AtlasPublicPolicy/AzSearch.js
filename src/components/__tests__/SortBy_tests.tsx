import * as React from "react";
import * as TestUtils from "react-dom/test-utils";
import SortBy from "../SortBy";

const fields = [
    { displayName: "Relevance", orderbyClause: "" },
    { displayName: "Beds", orderbyClause: "beds desc" },
    { displayName: "Baths", orderbyClause: "baths desc" }
];
const orderby = "";

describe("<SortBy/>", () => {
    it("renders against dummy snapshot", () => {

        expect(TestUtils.createRenderer().render(
            <SortBy  beforeFirstRequest={false} onSortChange={function(foo){}} fields={fields} css={null} orderby={orderby} />
        )).toMatchSnapshot();
    });
});