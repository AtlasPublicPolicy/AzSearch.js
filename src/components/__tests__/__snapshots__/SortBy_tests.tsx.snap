// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SortBy/> renders against dummy snapshot 1`] = `
<div
  className="sorting__sortyBy panel-body"
>
  <div
    className="searchResults__facetHeader-container panel-heading"
  >
    <h4
      className="searchResults__facetHeader panel-title"
    >
      <a
        className="searchResults__facetHeader-link"
        data-toggle="collapse"
      >
        <span
          aria-hidden="true"
          className="searchResults__facetHeader-icon--open indicator glyphicon glyphicon-triangle-bottom"
        />
         Sort By
      </a>
    </h4>
  </div>
  <div
    className="searchResults__facetControl-container panel-collapse collapse in"
  >
    <div
      className="searchResults__facetControl-list list-group"
    >
      <select
        className="sorting__sortyByControl list-group-item"
        onChange={[Function]}
      >
        <option
          selected={true}
          value=""
        >
          Relevance
        </option>
        <option
          selected={false}
          value="beds desc"
        >
          Beds
        </option>
        <option
          selected={false}
          value="baths desc"
        >
          Baths
        </option>
      </select>
    </div>
  </div>
</div>
`;
